src/fds/tests/test_stream_tracker.cpp

```diff
@@ -111,6 +113,45 @@ TEST_F(StreamTrackerTest, ForceRealloc) {
     EXPECT_EQ(get_mem_size(), prev_size * 2);
 }
 
+TEST_F(StreamTrackerTest, Rollback) {
+    static std::random_device s_rd{};
+    static std::default_random_engine s_engine{s_rd()};
+    std::uniform_int_distribution< int > gen{0, 999};
+
+    for (auto i = 0; i < 200; ++i) {
+        m_tracker.create(i, gen(s_engine));
+    }
+    EXPECT_EQ(m_tracker.active_upto(), 199);
+    EXPECT_EQ(m_tracker.completed_upto(), -1);
+    m_tracker.complete(0, 99);
+    EXPECT_EQ(m_tracker.active_upto(), 199);
+    EXPECT_EQ(m_tracker.completed_upto(), 99);
+
+    m_tracker.rollback(169);
```

Comment:
Should we test an invalid rollback as well?

Reply:
Let me add a test here, have to figure out a way to catch the exception and validate them.